gemstone/src/swapper/chainflip/broker/model.rs

```diff
@@ -0,0 +1,71 @@
+use alloy_primitives::U256;
+use num_bigint::BigUint;
+use serde::{Deserialize, Serialize};
+use serde_serializers::{deserialize_biguint_from_hex_str, serialize_biguint_to_hex_str};
+
+use crate::swapper::SwapperError;
+
+#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
+pub struct ChainflipAsset {
+    pub chain: String,
+    pub asset: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DepositAddressResponse {
+    pub address: String,
+    pub expiry_block: u64,
+    pub issued_block: u64,
+    pub channel_id: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub struct RefundParameters {
+    pub retry_duration: u32,
+    pub refund_address: String,
+    #[serde(deserialize_with = "deserialize_biguint_from_hex_str", serialize_with = "serialize_biguint_to_hex_str")]
+    pub min_price: BigUint,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DcaParameters {
+    pub number_of_chunks: u32,
+    pub chunk_interval: u32,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct VaultSwapExtraParams {
```

Comment:
Might want to rename to `VaultEvmSwapExtraParams` since they are different for Btc and Sol